<?xml version="1.0"?>
<launch>
    <!--<param name="use_sim_time"          value="true" />-->
    <arg name="info_topic"              value="/camera_info"/>
    <arg name="left_image_topic"        value="/wide/left/image_raw"/>
    <arg name="right_image_topic"       value="/wide/right/image_raw"/>
    <arg name="queue_size_img"          value="10"/>
    <arg name="my_ns"                   value="/wide"/>

    <group ns="$(arg my_ns)">

        <node pkg="undistort_images" type="publishCameraInfo" name="publishCameraInfo_left">
            <param name="topic/info"        value="$(arg info_topic)" />
            <param name="topic/img"         value="$(arg left_image_topic)" />
            <param name="camfile"           value="left_wide" />
            <param name="queue_size/img"    value="$(arg queue_size_img)" />
            <param name="cut"               value="true" />
            <!--ignored if cut is set to false-->
            <param name="x_offset"          type="int" value="0" />
            <param name="y_offset"          type="int" value="50" />
            <param name="width"             type="int" value="640" />
            <param name="height"            type="int" value="322" />
        </node>

        <node pkg="undistort_images" type="publishCameraInfo" name="publishCameraInfo_right">
            <param name="topic/info"        value="$(arg info_topic)" />
            <param name="topic/img"         value="$(arg right_image_topic)" />
            <param name="camfile"           value="right_wide" />
            <param name="queue_size/img"    value="$(arg queue_size_img)" />
            <param name="cut"               value="true" />
            <!--ignored if cut is set to false-->
            <param name="x_offset"          type="int" value="0" />
            <param name="y_offset"          type="int" value="50" />
            <param name="width"             type="int" value="640" />
            <param name="height"            type="int" value="322" />

        </node>

        <node pkg="stereo_image_proc" type="stereo_image_proc" name="stereo_image_proc">
            <remap from="/stereo/left/image_raw"    to="$(arg left_image_topic)" />
            <remap from="/stereo/right/image_raw"   to="$(arg right_image_topic)" />
            <remap from="/stereo/left/camera_info"  to="$(arg my_ns)/left$(arg info_topic)" />
            <remap from="/stereo/right/camera_info" to="$(arg my_ns)/right$(arg info_topic)" />
        </node>

    </group>
</launch>
